### DISCUSSION TITLE
/getting-started/compile.html

### DISCUSSION THREAD

**Discussion by giscus** [goravel/goravel#386] [Announcements]
# /getting-started/compile.html

[[toc]] 常规编译 部署服务器 部署时需要将下列文件与文件夹上传至服务器： 静态编译 常规编译后的包还需要依赖部署环境的支持，静态编译出的文件可以任意放到指定平台下运行，不需要环境配置。 交叉编译 编译有平台区分，需要根据部署情况，选择匹配的编译方式。 Docker Goravel 默认自带 Dockerfile 与 docker-compose...

https://www.goravel.dev/zh/getting-started/compile.html

**Comment by xx1196** [comment]
ERR_EMPTY_RESPONSE in the browser, but in docker can see the output wit 127.0.0.1:3000

**Reply by hwbrzzl** [reply]
If you are using docker, you have set APP_HOST=0.0.0.0

**Comment by jxgame** [comment]
打包的文件不能执行要怎么弄呢？
通过File命令查看，以前打包的是：
ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, not stripped

**Reply by jxgame** [reply]
今天打包的，用File命令看到的是：Mach-O 64-bit executable

**Reply by jxgame** [reply]
用这个go build --ldflags "-extldflags -static" -o main . 编译的，后来用// 编译 Linux 环境
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build .

**Reply by jxgame** [reply]
后面一种办法是可用的。
